splunk_add_on_ucc_framework/install_libs.py

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import os
+import shutil
+import stat
+from pathlib import Path
+from typing import Sequence
+
+
+def install_libs(
+    logger: logging.Logger, path: str, ucc_lib_target: str, python_binary_name: str
+):
+    logger.info(f"  Checking for requirements in {path}")
+    if os.path.exists(os.path.join(path, "lib", "requirements.txt")):
+        logger.info("  Uses common requirements")
+        _install_libraries(
+            logger,
+            os.path.join(path, "lib", "requirements.txt"),
+            ucc_lib_target,
+            python_binary_name,
+        )
+    elif os.path.exists(
+        os.path.join(os.path.abspath(os.path.join(path, os.pardir)), "requirements.txt")
+    ):
+        logger.info("  Uses common requirements")
+        _install_libraries(
+            logger,
+            os.path.join(
+                os.path.abspath(os.path.join(path, os.pardir)), "requirements.txt"
+            ),
+            ucc_lib_target,
+            python_binary_name,
+        )
+    else:
+        logger.info("  Not using common requirements")
+
+    packages_to_remove = ["setuptools", "bin", "pip", "distribute", "wheel"]
+    _remove_package_from_installed_path(
+        logger,
+        ucc_lib_target,
+        packages_to_remove,
+    )
+
+    _remove_execute_bit(logger, ucc_lib_target)
+
+
+def _install_libraries(
```

Comment:
Why using inner functions syntax? Usually we do that when we use functions inside functions, and it makes harder to write unit tests.